packages/ag-grid-community/src/edit/editModelService.ts

```diff
@@ -38,16 +38,48 @@ export class EditModelService extends BeanStub implements NamedBean, IEditModelS
         }
     }
 
-    public getEditRow({ rowNode }: EditRowPosition): EditRow | undefined {
-        return rowNode && this.edits.get(rowNode);
+    public getEditRow({ rowNode }: EditRowPosition, params: GetEditsParams = {}): EditRow | undefined {
+        let edits = rowNode && this.edits.get(rowNode);
+        if (!edits && params.checkSiblings) {
+            const pinnedSibling = (rowNode as RowNode).pinnedSibling;
+            if (params.checkSiblings && pinnedSibling) {
+                edits = this.getEditRow({ rowNode: pinnedSibling });
+            }
+        }
+        return edits;
     }
 
-    public getEdit(position: EditPosition): EditValue | undefined {
-        return position.column && this.getEditRow(position)?.get(position.column);
+    public getEditRowDataValue({ rowNode }: Required<EditRowPosition>, { checkSiblings }: GetEditsParams = {}): any {
+        const editRow = this.getEditRow({ rowNode });
+        const pinnedSibling = (rowNode as RowNode).pinnedSibling;
+        const siblingRow = checkSiblings && pinnedSibling && this.getEditRow({ rowNode: pinnedSibling });
+
+        if (!editRow && !siblingRow) {
+            return rowNode.data;
+        }
+
+        const data: any = Object.assign({}, rowNode.data);
+
+        const applyEdits = (edits: EditRow, data: any) =>
+            Array.from(edits.entries()).forEach(([column, { newValue }]) => {
```

Comment:
```suggestion
            edits.forEach(({ newValue }, column) => {
```
`entries` should be failing lint